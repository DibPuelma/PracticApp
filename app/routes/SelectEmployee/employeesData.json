{
  "total": 4,
  "images": ["http://people.csail.mit.edu/sanchez/daniel.jpg",
    "http://people.csail.mit.edu/sanchez/daniel.jpg",
    "http://people.csail.mit.edu/sanchez/daniel.jpg",
    "http://people.csail.mit.edu/sanchez/daniel.jpg"]

}
