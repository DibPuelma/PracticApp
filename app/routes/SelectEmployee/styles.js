import {
  StyleSheet,
  Dimensions
} from 'react-native';

module.exports = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 30,
    marginLeft: 0,
    marginTop: 30
  },
  card:Â {
    flexDirection: 'row',
    justifyContent: 'center',
    flexWrap: 'wrap',
    margin: 15,
    borderWidth: 0.5,
    borderRadius: 5,
    borderColor: '#E6E6E6',
    backgroundColor: '#E6E6E6',
    elevation: 3,
    paddingTop: 20,
    paddingBottom: 20
  },
  imageContainer: {
    width: 140,
    height: 140,
    margin: 5,
    borderColor: '#C6C6C6',
    borderWidth: 0.5,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#C6C6C6'
  },
  textChoice: {
    fontSize: 20,
    textAlign: 'center',
  },
  image: {
    height: 110,
    width: 110
  },
  title: {
    fontSize: 24,
    textAlign: 'center',
    marginTop: 10
  },
  employeeName: {
    textAlign: 'center'
  }
});
