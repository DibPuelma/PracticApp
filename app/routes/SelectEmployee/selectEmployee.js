import React, { Component } from 'react';
import {
  Text,
  Image,
  ScrollView,
  TouchableHighlight,
  StyleSheet,
  Dimensions,
  View,
  BackAndroid,
  Alert
} from 'react-native';

import styles from './styles';
import employeesData from './employeesData.json';
import backButtonHandler from '../../lib/backButtonHandler';

export default class SelectEmployee extends Component{

  constructor(props){
    super(props);
    this._backToPrevious = this._backToPrevious.bind(this);
    this.singletonBackButtonHandler = backButtonHandler.getInstance();
  }
  componentWillMount(){
    this._addBackEvent();
  }

  componentWillUnmount() {
    this._removeBackEvent();
  }

  _addBackEvent() {
    BackAndroid.addEventListener('hardwareBackPress', this._backToPrevious);
    this.singletonBackButtonHandler.addFunction(this._backToPrevious);
  }

  _removeBackEvent() {
    BackAndroid.removeEventListener('hardwareBackPress', this._backToPrevious);
    this.singletonBackButtonHandler.removeFunction(this._backToPrevious);
  }

  _backToPrevious() {
    console.log("Poping SE");
    Alert.alert(
      '¿Quieres salir de la encuesta?',
      'Recuerda que con solo 1 minuto de tu tiempo puedes ayudar a esta tienda a mejorar su servicio. Además estarás participando por premios mensuales',
      [
        {text: 'No', style: 'cancel'},
        {text: 'Sí', onPress: () => this.props.navigator.replace({id: 'scanner'})},
      ]
    )
    return true; // This is important to prevent multiple calls
  }

  render(){
    store = this.props.data.data.data;
    switch (store) {
      case 'encuesta1':
      return (
        <ScrollView contentContainerStyle={styles.container}>
        <Text style={styles.title}> ¿Quién lo atendió hoy? </Text>

        <View style={styles.card}>

        {employeesData.images.map((source, i) => (
          <TouchableHighlight style={styles.imageChoice} key={i} onPress={() => this._buttonPressed((i+1), true)}>
          <Image style={styles.imageChoice}
          source={{uri: source}}
          />
          </TouchableHighlight>
        ))}

        <TouchableHighlight style={styles.textContainer}
        onPress={() => this._buttonPressed(0, true)}>
        <Text style={styles.textChoice}> No lo recuerdo </Text>
        </TouchableHighlight>

        <TouchableHighlight style={styles.textContainer}
        onPress={() => this._buttonPressed(0, false)}>
        <Text style={styles.textChoice}> No me atendieron </Text>
        </TouchableHighlight>

        </View>
        </ScrollView>
      );
      case 'encuesta2':
      return (
        <ScrollView>

        <TouchableHighlight onPress={() => this._buttonPressed(1, true)}>
        <Image style={{width: 150, height: 150}}
        source={require('../../images/user.png')}
        />
        </TouchableHighlight>

        <TouchableHighlight onPress={() => this._buttonPressed(2, true)}>
        <Image style={{width: 150, height: 150}}
        source={require('../../images/user.png')}
        />
        </TouchableHighlight>

        <TouchableHighlight onPress={() => this._buttonPressed(3, true)}>
        <Image style={{width: 150, height: 150}}
        source={require('../../images/user.png')}
        />
        </TouchableHighlight>

        <TouchableHighlight onPress={() => this._buttonPressed(4, true)}>
        <Image style={{width: 150, height: 150}}
        source={require('../../images/user.png')}
        />
        </TouchableHighlight>

        <TouchableHighlight style={{width: 150, height: 150}}
        onPress={() => this._buttonPressed(0, false)}>
        <Text style={{fontSize:20}}> No lo recuerdo </Text>
        </TouchableHighlight>

        <TouchableHighlight style={{width: 150, height: 150}}
        onPress={() => this._buttonPressed(0, false)}>
        <Text style={{fontSize:20}}> No me atendieron </Text>
        </TouchableHighlight>

        </ScrollView>
      );
      default:
      return(
        <Text> Código no es válido </Text>
      );
    }
  }
  _buttonPressed(employeeId, wasAtended){
    pollData = {wasAtended:wasAtended, store:this.props.data.data.data, employeeId:employeeId}
    this.props.navigator.push({id:'poll', pollData:pollData});
  }
}
