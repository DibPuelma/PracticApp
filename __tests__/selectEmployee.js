import 'react-native';
import React from 'react';
import SelectEmployee from '../app/routes/SelectEmployee/selectEmployee'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

storeData =Â {
  data:"encuesta1",
  type:"QR"
}
it('renders correctly', () => {
  const tree = renderer.create(
    <SelectEmployee codeData={storeData}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
