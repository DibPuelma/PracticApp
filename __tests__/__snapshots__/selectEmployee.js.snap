exports[`test renders correctly 1`] = `
<ScrollView
  style={
    Object {
      "flex": 1,
      "marginBottom": 0,
      "marginLeft": 0,
      "marginRight": 0,
      "marginTop": 72,
      "paddingTop": 30,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 30,
        "textAlign": "center",
      }
    }>
     ¿Quién lo atendió hoy? 
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#E6E6E6",
        "borderColor": "#E6E6E6",
        "borderRadius": 5,
        "borderWidth": 0.5,
        "elevation": 3,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "margin": 20,
        "paddingBottom": 20,
        "paddingTop": 20,
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "height": 120,
            "margin": 5,
            "width": 120,
          },
        ]
      }
      testID={undefined}>
      <Image
        source={
          Object {
            "uri": "http://people.csail.mit.edu/sanchez/daniel.jpg",
          }
        }
        style={
          Object {
            "height": 120,
            "margin": 5,
            "width": 120,
          }
        } />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "height": 120,
            "margin": 5,
            "width": 120,
          },
        ]
      }
      testID={undefined}>
      <Image
        source={
          Object {
            "uri": "http://people.csail.mit.edu/sanchez/daniel.jpg",
          }
        }
        style={
          Object {
            "height": 120,
            "margin": 5,
            "width": 120,
          }
        } />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "height": 120,
            "margin": 5,
            "width": 120,
          },
        ]
      }
      testID={undefined}>
      <Image
        source={
          Object {
            "uri": "http://people.csail.mit.edu/sanchez/daniel.jpg",
          }
        }
        style={
          Object {
            "height": 120,
            "margin": 5,
            "width": 120,
          }
        } />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "height": 120,
            "margin": 5,
            "width": 120,
          },
        ]
      }
      testID={undefined}>
      <Image
        source={
          Object {
            "uri": "http://people.csail.mit.edu/sanchez/daniel.jpg",
          }
        }
        style={
          Object {
            "height": 120,
            "margin": 5,
            "width": 120,
          }
        } />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#C6C6C6",
            "borderColor": "#C6C6C6",
            "borderWidth": 0.5,
            "height": 120,
            "justifyContent": "center",
            "margin": 5,
            "width": 120,
          },
        ]
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "textAlign": "center",
          }
        }>
         No lo recuerdo 
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#C6C6C6",
            "borderColor": "#C6C6C6",
            "borderWidth": 0.5,
            "height": 120,
            "justifyContent": "center",
            "margin": 5,
            "width": 120,
          },
        ]
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "textAlign": "center",
          }
        }>
         No me atendieron 
      </Text>
    </View>
  </View>
</ScrollView>
`;
